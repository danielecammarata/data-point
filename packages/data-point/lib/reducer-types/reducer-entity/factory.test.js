/* eslint-env jest */

const factory = require("./factory");
const createReducer = require("../index").create;
const EntityReducer = require("../../entity-types/entity-transform");

test("isType", () => {
  expect(factory.isType("a")).toBe(false);
  expect(factory.isType({})).toBe(false);
  expect(
    factory.isType({
      isEntityInstance: true
    })
  ).toBe(true);
});

describe("create", () => {
  test("default create", () => {
    const reducer = factory.create(createReducer, EntityReducer("foo", "$"));
    expect(reducer).toMatchSnapshot();
  });
});
