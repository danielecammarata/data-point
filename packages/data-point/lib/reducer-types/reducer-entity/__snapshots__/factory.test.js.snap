// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create default create 1`] = `
ReducerEntity {
  "asCollection": false,
  "entity": reducer:foo {
    "entityType": "reducer",
    "id": "reducer:foo",
    "isEntityInstance": true,
    "name": "foo",
    "params": Object {},
    "resolve": [Function],
    "value": ReducerPath {
      "asCollection": false,
      "body": [Function],
      "name": "$",
      "type": "ReducerPath",
      Symbol(reducer): true,
    },
  },
  "entityType": "reducer",
  "hasEmptyConditional": false,
  "id": "reducer:foo",
  "name": "foo",
  "spec": reducer:foo {
    "entityType": "reducer",
    "id": "reducer:foo",
    "isEntityInstance": true,
    "name": "foo",
    "params": Object {},
    "resolve": [Function],
    "value": ReducerPath {
      "asCollection": false,
      "body": [Function],
      "name": "$",
      "type": "ReducerPath",
      Symbol(reducer): true,
    },
  },
  "type": "ReducerEntity",
}
`;
